lib/DataGridAbstractExtension.php

```diff
@@ -20,167 +18,143 @@
 {
     /**
-     * All column types extensions provided by data grid extension.
-     *
-     * @var array
+     * @var ColumnTypeExtensionInterface[][]
      */
     protected $columnTypesExtensions;
 
     /**
-     * All column types provided by extension.
-     *
-     * @var array
+     * @var ColumnTypeInterface[]
      */
     protected $columnTypes;
 
-    /**
-     * Returns a column type by id (all column types mush have unique id).
-     *
-     * @param string $id The identity of the column type
-     * @return \FSi\Component\DataGrid\Column\ColumnTypeInterface The column type
-     * @throws \FSi\Component\DataGrid\Exception\DataGridException if the given column type is not a part of this extension
-     */
-    public function getColumnType($type)
+    public function getColumnType(string $type): ColumnTypeInterface
     {
-        if (!isset($this->columnTypes)) {
+        if (null === $this->columnTypes) {
             $this->initColumnTypes();
         }
 
-        if (!isset($this->columnTypes[$type])) {
-            throw new DataGridException(sprintf('The column type "%s" can not be loaded by this extension', $type));
+        if (!array_key_exists($type, $this->columnTypes)) {
+            throw new DataGridException(sprintf(
+                'The column type "%s" can not be loaded by this extension',
+                $type
+            ));
         }
 
         return $this->columnTypes[$type];
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function hasColumnType($type)
+    public function hasColumnType(string $type): bool
     {
-        if (!isset($this->columnTypes)) {
+        if (null === $this->columnTypes) {
             $this->initColumnTypes();
         }
 
-        return isset($this->columnTypes[$type]);
+        return array_key_exists($type, $this->columnTypes);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function hasColumnTypeExtensions($type)
+    public function hasColumnTypeExtensions(string $type): bool
     {
-        if (!isset($this->columnTypesExtensions)) {
+        if (null === $this->columnTypesExtensions) {
             $this->initColumnTypesExtensions();
         }
 
-        return isset($this->columnTypesExtensions[$type]);
+        return array_key_exists($type, $this->columnTypesExtensions);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getColumnTypeExtensions($type)
+    public function getColumnTypeExtensions(string $type): array
     {
-        if (!isset($this->columnTypesExtensions)) {
+        if (null === $this->columnTypesExtensions) {
             $this->initColumnTypesExtensions();
         }
 
-        if (!isset($this->columnTypesExtensions[$type])) {
-            throw new DataGridException(sprintf('Extension for column type "%s" can not be loaded by this data grid extension', $type));
+        if (!array_key_exists($type, $this->columnTypesExtensions)) {
```

Comment:
Out of curiosity - why replace all `isset` with `array_key_exists`?

Reply:
In most cases the difference doesn't matter but `array_key_exists` is more generic. It only checks if key exists in array. `isset` also checks if the value under the key is not `null`.